vars/pipelineDeploy.groovy

```diff
@@ -180,6 +207,55 @@ void call() {
                     }
                 }
             }
+            stage('Backup') {
+                when {
+                    expression {
+                        env.NET_NAME == 'testnet'
+                    }
+                }
+                steps {
+                    script {
+                        veganet('chainstorecopy vegalogcopy')
+                    }
+                }
+            }
+            stage('Store checkpoint') {
+                when {
+                    expression {
+                        env.NET_NAME == 'testnet'
+                    }
+                }
+                steps {
+                    withCredentials([sshCredentials]) {
+                        script {
+                            netSsh = { command ->
+                                return 'ssh -t -i $PSSH_KEYFILE $PSSH_USER@n04.$NET_NAME.vega.xyz "sudo' + command + '"'
+                            }
+                            newestFile = sh (
+                                script: netSsh 'ls -t /home/vega/.local/state/vega/node/checkpoints/ | head -n 1',
+                                returnStdout: true
+                            ).trim()
+                            version = sh (
+                                script: netSsh "/home/vega/current/vega version | awk '{print $3}'"
+                            )
+                            sh netSsh "cp  /home/vega/.local/state/vega/node/checkpoints/${newestFile} /tmp/${newestFile}; chown `whoami`:`whoami` /tmp/${newestFile}"
+                            sh 'scp -i $PSSH_KEYFILE $PSSH_USER@n04.$NET_NAME.vega.xyz:/tmp/${newestFile} .'
+                            sh "mkdir -p checkpoint-store/Testnet-1/${version}"
+                            sh "mv ${newestFile} checkpoint-store/Testnet-1/${version}/"
+                            dir('checkpoint-store'){
+                                sh "vegatools checkpoint --file 'Testnet-1/${version}/${newestFile}' --out 'Testnet-1/${version}/${newestFile}.json'"
+                                sshagent(credentials: ['vega-ci-bot']) {
+                                    sh 'git config --global user.email "vega-ci-bot@vega.xyz"'
```

Comment:
Is jenkins able to checkout to the main branch or it will checkout to the specific commit?

Reply:
It goes to the main branch always -> https://github.com/vegaprotocol/jenkins-shared-library/pull/169/files#diff-91494a3a0a2f7511e3f572efd7eef6686f778cb908cad29732bcea055134ae10R127 and I think this is good behaviour to don't cause any conflicts